hrms/hr/doctype/shift_type/shift_type.py

```diff
@@ -42,6 +43,31 @@ def unlinked_checkins_exist(self):
 			{"shift": self.name, "attendance": ["is", "not set"], "skip_auto_attendance": 0, "is_invalid": 0},
 		)
 
+	def validate_circular_shift(self):
+		start = get_time(self.start_time)
+		end = get_time(self.end_time)
+		shift_start = datetime.combine(getdate(), start)
+		if start < end:
+			shift_end = datetime.combine(getdate(), end)
+		elif start > end:
+			shift_end = datetime.combine(add_days(getdate(), 1), end)
+		else:
+			frappe.throw(
+				title=_("Invalid Shift Times"),
+				msg=_("Start time and end time cannot be same."),
+			)
+		total_minutes = (
+			time_diff_in_minutes(shift_end, shift_start)
+			+ self.allow_check_out_after_shift_end_time
+			+ self.begin_check_in_before_shift_start_time
+		)
+
+		if total_minutes >= 1440:
+			frappe.throw(
+				title=_("Invalid Shift Times"),
+				msg=_("Please change shift times or buffers to avoid overlap."),
```

Comment:
The message is unclear. Point to the exact fields we need to change. "buffers" is unclear

Reply:
Also, maybe explicitly say, shift timings are overlapping with itself
Overlapping with what is ambiguous